### PR TITLE
Remove teachercon mapping

### PR DESCRIPTION
**Pull request by clareconstantine**
Partners that attended teachercon last year were still seeing their application dashboard with the experience they had last year - this is a minimal change to get them back to the typical experience (with the Registered Workshop column, for example). We need to follow up and remove all remaining teachercon-related code.

### CODE REVIEW

File: dashboard/lib/pd/application/regional_partner_teachercon_mapping.rb [hunk 1]
```diff
@@ -9,23 +9,7 @@ module RegionalPartnerTeacherconMapping
...
-      'Twin Cities Public Television' => TC_ATLANTA,
-      "UNH STEM Teachers' Collaborative" => TC_PHOENIX,
-      'Union Station' => TC_PHOENIX,
-      'University of Nebraska' => TC_ATLANTA,
-      'University of Rhode Island' => TC_PHOENIX,
-      'West Virginia University' => TC_ATLANTA,
-      'WNY STEM Hub' => TC_PHOENIX,
-      'Women in Technology' => TC_ATLANTA
-    }.freeze
+    REGIONAL_PARTNER_TC_MAPPING = {}.freeze
```
Comment (islemaster) [inline dashboard/lib/pd/application/regional_partner_teachercon_mapping.rb:12]: :grinning: Excited to see these hard-coded mappings go away!

**Review by islemaster — APPROVED** [review]

**Review by hacodeorg — APPROVED** [review]
Do we need to keep the TeacherCon tests now that there is no TeacherCon left?

**Review by hacodeorg — APPROVED** [review]
Sorry, I accidentally click on re-request myself to review your PR 🌀

**Comment by clareconstantine** [comment]
@hacodeorg good point - there is a lot of teachercon-related code still, including tests. This is the minimum change to unblock partners today, so I left in tests for methods that still exist, even though we need to follow up and remove those methods later. I didn't want to make more changes than I have time to thoroughly test right now.

**Pull request merged**